core/utils.py

```diff
@@ -91,6 +92,28 @@ class ExplorationConversionError(Exception):
     pass
 
 
+def open_file(
+        filename: str, mode: str, encoding: Union[str, None]='utf-8',
+        newline: Union[str, None]=None) -> io.TextIOWrapper:
+    """Open file and return a corresponding file object.
+
+    Args:
+        filename: str. The file to be opened.
+        mode: str. Mode in which the file is opened.
+        encoding: str. Encoding in which the file is opened.
+        newline: None|str. Controls how universal newlines work.
```

Comment:
This should be clearer, it doesn't explain how to use the argument.

Also, is this argument actually used by any caller in the codebase? If not, you can remove it.

Reply:
![2022 01 22-11 03 38 screenshot](https://user-images.githubusercontent.com/63491234/150626312-6a0cee8f-0d6a-4030-9b1b-efc1f0a970e8.png)

These arguments are helpful especially when you are using some language server as you can see above . It gives more brief information about what the function is actually about and what it's arguments does .

Reply:
I don't think you've answered my second question, or addressed my first one properly. I still don't know, as a developer, how to decide on what value to pass in for the last argument, just from reading the docstring.

Reply:
Sorry for the late reply @seanlip !! The argument lines above is directly taken from the previous open-file function from python_utils.py ( python 2 / python 3 ) . I agree with you that the argument doesn't give full understanding of what it actually does . I will apply new changes to them .

Reply:
You still haven't answered the second question in my original comment.

Reply:
Yes these arguments are surely being used in the codebase . There are instances of open_file where we want a different encoding or differnt io stream ( binary / raw ) .

Reply:
OK, thanks. If you could, please point to some usages explicitly (using the github link). That will help with checking whether the docstring is suitable or not.